### PR TITLE
Upgrade to Rector 0.7.19

### PR DESCRIPTION
**Pull request by TomasVotruba**
I was working on `vendor/bin/rector dump-rectors`  command [being opened for public](https://github.com/rectorphp/rector/pull/3265),
while I've noticed, this package is still on old Rector 0.6.

I think I might give you helping hand, since I made many BC breaks in version 0.7.

Hopefully it helps :)

### CODE REVIEW

**Comment by TomasVotruba** [comment]
3 hours later...

Uff, that was lot of work, but I think it should work now.

<br>

Would you mind PR for better coding standard? Current setup is very limited and I had to do lot of stuff manually (I'm to spoiled by strong coding standards :D )

**Comment by TomasVotruba** [comment]
Ready to merge :+1: 

Feel free to ask anything, I'll try to explain

File: phpunit.xml [hunk 1]
```diff
@@ -3,7 +3,6 @@
         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
         bootstrap="tests/bootstrap.php"
         colors="true"
-        stopOnFailure="true"
```
Comment (TomasVotruba) [inline phpunit.xml:6]: This should make testing faster, as all the errors are shown. Picking low handing fruit is easier than to be forced to solve first failed case.

File: src/NodeFactory/InitializeArgumentsClassMethodFactory.php [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+use PHPStan\PhpDocParser\Ast\PhpDoc\ParamTagValueNode;
+use PHPStan\Type\Type;
+use Rector\AttributeAwarePhpDoc\Ast\PhpDoc\AttributeAwareParamTagValueNode;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo;
+use Rector\Core\PhpParser\Node\NodeFactory;
+use Rector\NodeNameResolver\NodeNameResolver;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\StaticTypeMapper\StaticTypeMapper;
+
+final class InitializeArgumentsClassMethodFactory
```
Comment (TomasVotruba) [inline src/NodeFactory/InitializeArgumentsClassMethodFactory.php:27]: Rector that uses this services was buggy, so this class was extracted to better isolate to responsibility.

File: src/PhpStan/Rules/AddSeeDocBlockForRectorClass.php [hunk 3]
```diff
@@ -59,12 +59,15 @@ public function processNode(Node $node, Scope $scope): array
...
 
-        if (!$classReflection->isSubclassOf(PhpRectorInterface::class) || in_array($fullyQualifiedClassName, self::$allowedClassesWithNonSeeDocBlock, true)) {
+        if (!$classReflection->isSubclassOf(PhpRectorInterface::class)) {
             return [];
         }
 
-        $docComment = $node->getDocComment();
+        if (in_array($fullyQualifiedClassName, self::$allowedClassesWithNonSeeDocBlock, true)) {
+            return [];
+        }
```
Comment (TomasVotruba) [inline src/PhpStan/Rules/AddSeeDocBlockForRectorClass.php:68]: Early return split

File: tests/config/typo3_rectors.yaml [hunk 4]
```diff
@@ -1,13 +1,2 @@
...
-    - { resource: "../../config/typo3-86.yaml" }
-    - { resource: "../../config/typo3-87.yaml" }
-    - { resource: "../../config/typo3-90.yaml" }
-    - { resource: "../../config/typo3-93.yaml" }
-    - { resource: "../../config/typo3-94.yaml" }
-    - { resource: "../../config/typo3-95.yaml" }
-    - { resource: "../../config/typo3-100.yaml" }
-    - { resource: "../../config/typo3-101.yaml" }
-    - { resource: "../../config/typo3-102.yaml" }
+    - { resource: "../../config/*" }
```
Comment (TomasVotruba) [inline tests/config/typo3_rectors.yaml:2]: Symfony glob tip: https://symfony.com/blog/new-in-symfony-3-3-import-config-files-with-glob-patterns

**Comment by sabbelasichon** [comment]
@TomasVotruba This is totally insane. Thanks a lot. I try to compensate your effort as much as i can.

**Pull request merged**